File: src/test/java/io/gaegul/ch01/RubberDuckTest.java
package io.gaegul.ch01;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import io.gaegul.util.ConsoleIOTest;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
@DisplayName("오리 시뮬레이션 게임 - 러버덕은")
class RubberDuckTest extends ConsoleIOTest {

	@Test
	void 삑삑소리를_낼_수_있다() {
		RubberDuck duck = new RubberDuck();
		duck.quack();
		assertThat(output()).isEqualTo("삑삑~");
	}

	@Test
	void 헤엄을_칠_수_있다() {
		Duck duck = new RubberDuck();
		boolean result = duck.swim();
		assertThat(result).isTrue();
	}

	@Nested
	class display_메서드를_실행하면 {

		@Test
		void 러버덕을_출력한다() {
			Duck duck = new RubberDuck();
			duck.display();
			assertThat(output()).isEqualTo("러버덕");
		}

	}
}

File: src/test/kotlin/io/gaegul/ch01/RubberDuckTest.kt
package io.gaegul.ch01

import io.gaegul.utils.ConsoleIOTest
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test

@DisplayName("오리 시뮬레이션 게임 - 러버덕은")
class RubberDuckTest: ConsoleIOTest() {

    private val duck: RubberDuck
        get() {
            return RubberDuck()
        }

    @Test
    fun `삑삑 소리를 낼 수 있다`() {
        duck.performQuack()
        assertThat(output()).isEqualTo("삑삑~")
    }

    @Test
    fun `혜엄을 칠 수 있다`() {
        val result = duck.swim()
        assertThat(result).isTrue()
    }

    @Test
    fun `날 수 없다`() {
        assertThatThrownBy { duck.performFly() }
            .isInstanceOf(UnsupportedOperationException::class.java)
    }

    @Test
    fun `display()를 실행하면 러버덕을 출력한다`() {
        duck.display()
        assertThat(output()).isEqualTo("러버덕")
    }

}